### ISSUE TITLE
Accuracy degradation after exporting model to onnx

### ISSUE THREAD

**Issue by wibbn** [openvinotoolkit/nncf#1590]
Hi!

I faced a problem with model quality after exporting model to onnx and inference with openvino. Model shows appropriate and fine accuracy when I just run it with torch. 

I'm using nncf for QAT where my nncf config looks like that:
```python
nncf_config = {
    'input_info': [
        {
            'sample_size': [1, 512, 4],
            "type": "long",
            "keyword": "bbox"
        },
        {
            'sample_size': [1, 3, 1000, 1000],
            "type": "float",
            "keyword": "pixel_values"
        },
        {
            'sample_size': [1, 512],
            "type": "long",
            "keyword": "input_ids"
        }
    ],
    'compression': {
        'algorithm': 'quantization',
        'ignored_scopes': [
            '{re}.*CustomLogWrapper[log_wrapper_function]*',
            '{re}.*LayoutLMv3Output[output]*',
        ]
    }
}
```

I'm loading pretrained model:
```python
nncf_config = NNCFConfig.from_dict(nncf_config)
nncf_config = register_default_init_args(
    nncf_config, init_dl, device=device
)
model =  MyModel()
compression_ctrl, model = create_compressed_model(model, nncf_config)
model = load_weights(model, cfg.MODEL.weights_path, device=device)
model.to(device)
```

Then training process flows just fine and then I am exporting my model to checkpoint
```python
checkpoint = {
        'epoch': epoch,
        'score': score,
        'state_dict': model.state_dict(),
        'compression_state': compression_ctrl.get_compression_state(),
        'optimizer': optimizer.state_dict()
    }
    torch.save(checkpoint, checkpoint_path)
```

On this stage I can load the model to pytorch and see fine metrics
```python
model =  MyModel()
resuming_checkpoint = load_resuming_checkpoint(resuming_checkpoint_path)
model_state_dict, compression_state = extract_model_and_compression_states(resuming_checkpoint)
compression_ctrl, model = create_compressed_model(model, nncf_config, compression_state)
if model_state_dict is not None:
    load_state(model, model_state_dict, is_resume=True)
```

But when I try to export it to onnx and inference it with openvino metrics are completely inconsistent.
```python
export_kwargs = dict(
    input_names=['bbox', 'pixel_values', 'input_ids'],
    output_names=['last_hidden_state'],
    save_format='onnx_13'
)
compression_ctrl.export_model('nncf-model.onnx', **export_kwargs)

core = ov.Core()
raw_q_model = core.read_model('nncf-model.onnx', "AUTO")
ov_q_model = core.compile_model(raw_q_model, "CPU", config={"INFERENCE_NUM_THREADS": 6})

infer_request = ov_q_model.create_infer_request()
infer_request.infer(b)
res = torch.tensor(infer_request.output_tensors[0].data)
```

Сomparison of metrics:
```
Quant (torch)	score: 0.9661
Quant (OV)	score: 0.5545
Original (OV)	score: 0.9657
```

I also tried to use export to OV IR but it didn't change anything.

Here are my requirements:
```python
torch==1.13.1
onnx==1.12.0
onnxruntime==1.3.0
openvino==2022.2.0
nncf @ git+https://github.com/openvinotoolkit/nncf.git@b1e579611dcb9ee0a7d7bea9cd8f604d57ad15c6
```
There is a similar issue #1284, but solution there seems indispensable because I'm dealing with a transformer and a lot of layers.
Can you, please, tell what can be the reason of such behavior?

**Comment by MaximProshin** [comment]
@vshampor , please take a look

**Comment by vshampor** [comment]
Greetings, @wibbn!

It is actually pretty hard to tell what exactly went wrong without the model code or at least a minimal reproducer. Here are some things you can try to narrow down the issue:
1) Try exporting to ONNX without saving/loading the checkpoint to see if it changes anything.
2) NNCF should have created the `original_graph.dot` files and `compressed_graph.dot` files in the current working directory that contain the dumps of the internal representation of the model in NNCF. Verify these for basic consistency (i.e. all ops are there, the graph is not disjoint), and then compare the `compressed_graph.dot` (which will contain the nodes representing the quantizers set up by NNCF) with the ONNX model as seen in Netron, so that every quantizer that is seen in `compressed_graph.dot` is also there in the correct spot in the ONNX.
3) Try to export to ONNX QuantizeLinear-DequantizeLinear representation by setting [this](https://openvinotoolkit.github.io/nncf/#compression_oneOf_i0_oneOf_i0_export_to_onnx_standard_ops) option in your NNCF config to `true` and then check the model both in OV and in regular onnxruntime. If onnxruntime shows good metrics, then the bug is on the OV side somewhere. If onnxruntime shows bad metrics, then NNCF is probably at fault and we would like to see the model code or a minimal reproducer to determine your exact failure case that we don't seem to support.

**Comment by MaximProshin** [comment]
@wibbn , based on https://github.com/openvinotoolkit/nncf/issues/1610 the problem might be connected with an older OpenVINO version where some fault could be present. Can you please try OpenVINO 2022.3 (the last version at this moment) and let us know if you still the accuracy degradation after the exporting?

**Comment by MaximProshin** [comment]
I close it as I think it was resolved. Otherwise, feel free to contact us.
